### PR TITLE
Remove_trivial_connectivity_lines

### PR DESCRIPTION
**Pull request by shireenf-ibm**
#143

### CODE REVIEW

File: pkg/netpol/eval/check.go [hunk 1]
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
 
 // for connectivity considerations, when requesting allowed connections between 2 workload peers which are the same,
 // looking for 2 different pod instances, if exist (to avoid the trivial case of connectivity from pod to itself)
-func (pe *PolicyEngine) changePodPeerToAnotherPodObject(peer *k8s.PodPeer) {
+// returns true if different pod instances found
```
Comment (adisos) [inline pkg/netpol/eval/check.go:79]:
```suggestion
// if different pod instance was found - returns true and updates the input peer with another pod instance
```

File: pkg/netpol/eval/check.go [hunk 2]
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
...
 // expecting that srcPeer and dstPeer are in level of workloads (WorkloadPeer)
-func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, error) {
+func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, bool, error) {
+	isTrivial := false
 	if srcPeer.IsPeerIPType() && !dstPeer.IsPeerIPType() {
 		// assuming dstPeer is WorkloadPeer, should be converted to k8s.Peer
 		dstPodPeer, err := pe.convertWorkloadPeerToPodPeer(dstPeer)
 		if err != nil {
-			return nil, err
+			return nil, isTrivial, err
```
Comment (adisos) [inline pkg/netpol/eval/check.go:101]:
```suggestion
			return nil, false, err
```

File: pkg/netpol/eval/check.go [hunk 3]
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
...
+	return false
 }
 
 // AllAllowedConnectionsBetweenWorkloadPeers returns the allowed connections from srcPeer to dstPeer,
+// and a bool flag indicating if the allowed connections are trivial
+// i.e. Connection from deployment to itself when the deployment has only one replica.
 // expecting that srcPeer and dstPeer are in level of workloads (WorkloadPeer)
-func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, error) {
+func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, bool, error) {
+	isTrivial := false
```
Comment (adisos) [inline pkg/netpol/eval/check.go:96]:
```suggestion
```
move the definition of is`Trivial` to be before the call to `changePodPeerToAnotherPodObject` , where it is relevant

File: pkg/netpol/eval/check.go [hunk 4]
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
...
 		if srcPeer.String() == dstPeer.String() {
-			pe.changePodPeerToAnotherPodObject(dstPodPeer)
+			isTrivial = !pe.changePodPeerToAnotherPodObject(dstPodPeer)
 		}
-		return pe.allAllowedConnectionsBetweenPeers(srcPodPeer, dstPodPeer)
+		conn, err := pe.allAllowedConnectionsBetweenPeers(srcPodPeer, dstPodPeer)
+		return conn, isTrivial, err
 	}
-	return nil, fmt.Errorf("cannot have both srcPeer and dstPeer of IP types: src: %s, dst: %s", srcPeer.String(), dstPeer.String())
+	return nil, isTrivial, fmt.Errorf("cannot have both srcPeer and dstPeer of IP types: src: %s, dst: %s", srcPeer.String(), dstPeer.String())
```
Comment (adisos) [inline pkg/netpol/eval/check.go:131]:
```suggestion
	return nil, false, fmt.Errorf("cannot have both srcPeer and dstPeer of IP types: src: %s, dst: %s", srcPeer.String(), dstPeer.String())
```

File: pkg/netpol/eval/check.go [hunk 5]
```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
...
 	}
+	return false
 }
 
 // AllAllowedConnectionsBetweenWorkloadPeers returns the allowed connections from srcPeer to dstPeer,
+// and a bool flag indicating if the allowed connections are trivial
+// i.e. Connection from deployment to itself when the deployment has only one replica.
 // expecting that srcPeer and dstPeer are in level of workloads (WorkloadPeer)
-func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, error) {
+func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, bool, error) {
```
Comment (adisos) [inline pkg/netpol/eval/check.go:95]: does it handle identifying trivial connections if the input peers are pods and not workloads? do we have such tests?

**Review by adisos — CHANGES_REQUESTED** [review]
hi @shireenf-ibm , added a few comments.
The main question is - do we handle both the case of input workload resources, and the case of input pod resources?  please check and see if we have tests for both scenarios.
Also, I tried to check `tests\onlineboutique_with_replicas` and it seems this test contains both pods and deployments resources -- I think it should be split into the two cases separately..

**Comment by shireenf-ibm** [comment]
> Also, I tried to check `tests\onlineboutique_with_replicas` and it seems this test contains both pods and deployments resources -- I think it should be split into the two cases separately..

we already had the `tests/onlineboutique_with_replicas_and_variants` which includes only pods resources.
so I removed `pods.yaml` from `tests\onlineboutique_with_replicas`, so it will include deployment resources only .
and added both to the `connlist_test.go ` so we can view their connections list.

for `tests\onlineboutique_with_replicas` , since it doesn't include pods resources any more, I had to remove its `tests/onlineboutique_with_replicas/connections_map_with_replicas_output.txt` and eliminate the relevant test in `eval_test.go`, as it creats an empty `podsMap` now.

File: pkg/netpol/eval/check.go (hunk 5 above)
Comment (shireenf-ibm) [reply pkg/netpol/eval/check.go:95]: seems to me yes it is, 
does the `podlist ` of `tests/ipblockstest` meets the requirement?

**Pull request closed without merging**